### DISCUSSION TITLE
Klogging and Junit/Mockito

### DISCUSSION THREAD

**Discussion by Andrewsoares15** [klogging/klogging#200] [Q&A]
Does anyone have an example of how to mock a Klogging object using Mockito? I'm facing some issues

Information: I have a test and I need to verify with logger.warn was called

**Comment by mjstrasser** [comment]
Hi @Andrewsoares15

Would it work to copy the [`savedEvents()` helper function](https://github.com/klogging/klogging/blob/main/klogging/src/jvmTest/kotlin/io/klogging/Helpers.kt#L68) from Klogging tests? This function sets up Klogging configuration to use in a single test case.

An example of its use might be:

```kotlin
    @Test
    fun `check for warn log events()` {
        val events = savedEvents()

        callFunctionUnderTest()

        assertEquals(1, events.size)
        assertEquals(Level.WARN, events.first().level)
    }
```

(The assertions may be wrong: I use Kotest and am not familiar with JUnit assertions.)
